### DISCUSSION TITLE
Consistent 504 `FUNCTION_INVOCATION_TIMEOUT` Timeouts

### DISCUSSION THREAD

**Discussion by troymcginnis** [vercel/community#632] [Help]
Hey Vercel community 👋🏼

Just under a month ago, I posted a discussion about surpassing [Serverless Function Limits](https://github.com/vercel/community/discussions/596) on #596. Since then, I've been working on getting on-demand ISR setup on our app to, hopefully, reduce our Serverless Function usage.

The local tests were promising. We encountered some stale content while using Sanity's CDN API, so we flipped off the CDN and were happily receiving fresh data.

However, once I pushed the app to Vercel, I was met with constant 504 Timeouts when hitting pages that had not yet been generated. Various errors like:
```
[GET] /_next/data/eW9cEX-gFPQGvnHB7tTEz/en-US/article/nfl-betting-who-will-be-the-steelers-starting-quarterback-in-week-1-43874.json
17:02:33:93
2022-06-02T00:03:35.401Z d83ef2b4-d84d-4cf8-9ffb-4fd13b86f02d Task timed out after 60.01 seconds
```
began to litter our logs. App performance, for non-generated pages, suffered greatly (as they all timeout after 60s) while generated pages worked perfectly.

<img width="1915" alt="Screen Shot 2022-05-31 at 12 54 48 PM" src="https://user-images.githubusercontent.com/4299551/171522776-2f798450-28f4-4977-8f6a-d071020624fb.png">
<img width="1914" alt="Screen Shot 2022-06-01 at 5 05 37 PM" src="https://user-images.githubusercontent.com/4299551/171522863-59054330-98d0-4d45-b04a-447d98987e38.png">

While generated pages do appear to be responding quickly and causing no issues, once the app has been redeployed (certain content requires a full deploy to appear), we end up back where we started with the previously generated pages being blasted away, regular timeouts until pages have been generated, and a less than ideal app experience.

I've opened a support request with Vercel and am talking to Sanity help, but am hoping for any suggestions or thoughts from Vercel/Vercel Community.

Any thoughts?

**Comment by ghost** [comment]
Did you ever figure out what was wrong? I'm seeing a similar issue (#1657).

**Reply by ghost** [reply]
I resolved my issue by removing named imports for MUI Icons as mentioned [here](https://github.com/vercel/next.js/discussions/32686#discussioncomment-3219147) and [here](https://mui.com/material-ui/guides/minimizing-bundle-size/#development-environment).
